sdk/cosmosdb/cosmos/src/queryExecutionContext/ContinuationTokenManager/ParallelQueryContinuationTokenManager.ts

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { BaseContinuationTokenManager } from "./BaseContinuationTokenManager.js";
+import type { QueryResponseResult } from "./BaseContinuationTokenManager.js";
+import type {
+  CompositeQueryContinuationToken,
+  QueryRangeWithContinuationToken,
+} from "../../documents/ContinuationToken/CompositeQueryContinuationToken.js";
+import {
+  createCompositeQueryContinuationToken,
+  serializeCompositeToken,
+  parseCompositeQueryContinuationToken,
+  convertRangeMappingToQueryRange,
+} from "../../documents/ContinuationToken/CompositeQueryContinuationToken.js";
+import type { QueryRangeMapping } from "../queryRangeMapping.js";
+
+/**
+ * Manages continuation tokens for parallel queries using multi-range aggregation.
+ * Uses CompositeQueryContinuationToken for range tracking across multiple partitions.
+ * @internal
+ */
+export class ParallelQueryContinuationTokenManager extends BaseContinuationTokenManager {
+  private continuationToken: CompositeQueryContinuationToken;
+
+  constructor(
+    collectionLink: string,
+    initialContinuationToken?: string,
+    isUnsupportedQueryType: boolean = false,
+  ) {
+    super(collectionLink, isUnsupportedQueryType);
+    if (initialContinuationToken) {
+      this.initializeFromToken(initialContinuationToken);
+    }
+  }
+
+  protected initializeFromToken(token: string): void {
+    this.continuationToken = parseCompositeQueryContinuationToken(token);
+    this.ranges = this.continuationToken.rangeMappings || [];
+  }
+
+  public createContinuationToken(
+    pageSize: number,
+    _isResponseEmpty: boolean,
+    responseResult?: QueryResponseResult,
+  ): {
+    endIndex: number;
+    processedRanges: string[];
+    continuationToken?: string;
+  } {
+    // Process response data first if provided
+    if (responseResult) {
+      this.processResponseResult(responseResult);
+    }
+
+    this.removeExhaustedRangesFromRanges();
+    const result = this.processRanges(pageSize);
+
+    // Clean up processed data automatically
+    this.cleanProcessedData(result.processedRanges, result.endIndex);
```

Comment:
Why do we want to clean data here?

Reply:
this is an internal method to clean the internal map for supporting continuation pagination.